packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -673,11 +688,30 @@ class DevFS {
           case AssetKind.regular:
           case AssetKind.font:
           case null:
-            dirtyEntries[deviceUri] = entry.content;
-            syncedBytes += entry.content.size;
-            if (!bundleFirstUpload) {
-              assetPathsToEvict.add(archivePath);
-            }
+            final Future<DevFSContent?> pending = (() async {
+              if (entry.transformers.isEmpty || kind != AssetKind.regular) {
+                return entry.content;
+              }
+              return _assetTransformer.retransformAsset(
+                inputAssetKey: archivePath,
+                inputAssetContent: entry.content,
+                transformerEntries: entry.transformers,
+                workingDirectory: rootDirectory.path,
+              );
+            })();
+
+            pendingAssetBuilds.add(pending);
+            pending.then((DevFSContent? content) {
+              if (content == null) {
+                assetBuildFailed = true;
+                return;
+              }
+              dirtyEntries[deviceUri] = content;
+              syncedBytes += content.size;
+              if (!bundleFirstUpload) {
```

Comment:
for my understanding, why do we not evict this path the first upload?

Reply:
Good question. The only reference to `assetsPathToEvict` is here: https://github.com/flutter/flutter/blob/4e70bfae2bff2c272af7fb7a870695edd9f8cfb4/packages/flutter_tools/lib/src/run_hot.dart#L1207-L1215

where make some vm service call, `ext.flutter.evict`. Do you know where to find the code that process this call?

Reply:
Nevermind. I'm starting to figure this out, give me a sec.

Reply:
So this VM service call ends up here: https://github.com/flutter/flutter/blob/7a4c2465afe875355aaac3efee158084f13e6fc7/packages/flutter/lib/src/services/binding.dart#L250-L258

which indeed just evicts the asset from the framework's cache. However, I don't know if there is a specific reason to _not_ vacuously evict from the cache in first upload (other than perhaps performance).